.github/workflows/chef-test.yml

```diff
@@ -80,20 +277,40 @@ jobs:
 
   chef-kitchen-linux:
     runs-on: ubuntu-24.04
-    needs: [chef-lint-spec-test, chef-kitchen-matrix]
+    needs: [chef-lint-spec-test, chef-kitchen-matrix, build-package]
+    defaults:
+      run:
+        working-directory: 'deployments/chef'
     strategy:
       matrix: ${{ fromJSON(needs.chef-kitchen-matrix.outputs.linux-matrix) }}
       fail-fast: false
     steps:
       - name: Check out code
         uses: actions/checkout@v4
 
+      - uses: actions/download-artifact@v4
+        with:
+          path: /tmp
+
+      # Chef doesn't have an easy way to get names of files from unarchived packages
+      # It's simpler to manually unarchive and move the built artifacts to the
+      # cookbook files directory here. The files/ dir under the Chef deployment is a special
+      # directory in Chef, used for copying files from the cookbook to the host running
+      # Chef.
+      # Cookbook files reference: https://docs.chef.io/files/
+      - name: Extract artifacts
+        run: |
+          mkdir -p files
+          deb_path=$(find /tmp/deb-amd64-package/splunk-otel-collector*amd64.deb)
```

Comment:
For all three paths, `deb_path`, `rpm_path`, and `msiPath`, we're relying on the directories to only have a single file that matches the regex. This might fail in the future if built artifacts change names, or end up having more than one matching artifact.